rust/src/operations/delete.rs

```diff
@@ -0,0 +1,812 @@
+    // If returned count is not zero then append the file to be rewritten and removed from the log. Otherwise do nothing to the file.
+
+    let schema = snapshot.arrow_schema().unwrap();
+
+    //TODO: Move this to a function.
+    let table_partition_cols = snapshot
+        .current_metadata()
+        .ok_or(DeltaTableError::NoMetadata)
+        .unwrap()
+        .partition_columns
+        .clone();
+    let file_schema = Arc::new(ArrowSchema::new(
+        schema
+            .fields()
+            .iter()
+            .filter(|f| !table_partition_cols.contains(f.name()))
+            .cloned()
+            .collect(),
+    ));
+    let expr = logical_expr_to_physical_expr(expression, &schema);
+
+    let pruning_predicate = PruningPredicate::try_new(expr, schema.clone())?;
+    let files_to_prune = pruning_predicate.prune(snapshot).unwrap();
+    let files: Vec<&Add> = snapshot
+        .files()
+        .iter()
+        .zip(files_to_prune.into_iter())
+        .filter_map(|(action, keep)| if keep { Some(action) } else { None })
+        .collect();
+
+    // The expression only contains partition columns so the scan can be skipped.
+    // This assumes pruning works correctly for partitions
+    if expr_properties.partition_only {
+        return Ok((
+            Vec::new(),
+            files.into_iter().map(|x| x.to_owned()).collect(),
+        ));
+    }
+
+    let scan_start = Instant::now();
+    // Create a new delta scan plan with only files that have a record
+    let rewrite = find_files(
+        snapshot,
+        object_store.clone(),
+        &schema,
+        file_schema.clone(),
+        files,
+        state,
+        expression,
+    )
+    .await?;
+
+    metrics.scan_time_ms = Instant::now().duration_since(scan_start).as_millis();
+
+    let rewrite: Vec<Add> = rewrite.into_iter().map(|s| s.to_owned()).collect();
+    let parquet_scan = parquet_scan_from_actions(
+        snapshot,
+        object_store.clone(),
+        &rewrite,
+        &schema,
+        None,
+        state,
+        None,
+        None,
+    )
+    .await?;
+
+    // Apply the negation of the filter and rewrite files
+    let negated_expression = Expr::Not(Box::new(expression.clone()));
+    let physical_schema = file_schema.clone();
+    let logical_schema: DFSchema = file_schema.as_ref().clone().try_into()?;
+    let execution_props = ExecutionProps::new();
+
+    let predicate_expr = create_physical_expr(
+        &negated_expression,
+        &logical_schema,
+        &physical_schema,
+        &execution_props,
+    )?;
+    let filter: Arc<dyn ExecutionPlan> =
+        Arc::new(FilterExec::try_new(predicate_expr, parquet_scan.clone())?);
+
+    let write_start = Instant::now();
+
+    let add_actions = write_execution_plan(
+        snapshot,
+        state.clone(),
+        filter.clone(),
+        table_partition_cols.clone(),
+        object_store.clone(),
+        None,
+        None,
+        Some(writer_properties),
+    )
```

Comment:
```suggestion
    let add_actions = write_execution_plan(
        snapshot,
        state.clone(),
        filter.clone(),
        table_partition_cols.clone(),
        object_store.clone(),
        Some(snapshot.table_config().target_file_size() as usize),
        None,
        Some(writer_properties),
    )
```

I guess a better way would be to extract defaults from the table config within `write_execution_plan`.